_posts/en/newsletters/2020-04-29-newsletter.md

```diff
@@ -0,0 +1,265 @@
+      Unfortunately, not all full nodes receive the same transactions
+      even when everything is working perfectly and there are techniques
+      attackers like Mallory can use to [send different conflicting
+      transactions][corallo mempool not guaranteed] to different peers
+      (for example, sending the pinned preimage settlement transaction
+      to known miners but sending a different non-settlement transaction
+      with at least one of the same inputs to non-miner relay nodes).
+
+    - **Beg or pay for preimages:** The relay network could provide
+      [information about conflicts][harding reject] to transaction
+      submitters such as Bob so they wouldn't need to continuously
+      monitor relay themselves.  This still suffers from the problem of
+      bad actors such as Mallory using [targeted relay][corallo targeted
+      relay] to send different transactions to miners and non-miners.
+      Additionally Bob might be able to [pay][harding pay] miners or
+      other third party nodes for the preimage he needs, although this
+      requires some people run additional software and [might not be as
+      easy][zmn ptlcs] to do after the deployment of some proposed
+      upgrades to the LN protocol.
+
+    - **Settlement transaction anchor outputs:** Onchain settlement
+      transactions could be redesigned to spend their value to [anchor
+      outputs][topic anchor outputs] that could be [CPFP fee
+      bumped][topic cpfp] using [CPFP carve-out][topic cpfp carve out].
+      This would require those transactions to be larger (increasing
+      onchain fees) and presigned (reducing flexibility).  This would
+      only affect channels which are unilaterally closed while payments
+      are pending, which is already a situation that that can
+      significantly increase onchain costs and so is something users try
+      to avoid.  As of this writing, this appears to be the most
+      preferred solution.
+
+    Corallo labeled this a severe issue but noted its similar
+    consequences to another known issue related to fee management in
+    onchain LN transactions.  The existing issue (described in
+    [Newsletter #78][news78 anchor outputs]) is that commitment
+    transactions have their feerate set at the time the transaction is
+    signed, which might be days or weeks before they're broadcast to
+    Bitcoin relay nodes.  If the minimum feerate necessary to get a
+    transaction included in the next few blocks has increased
+    significantly since the transaction was last signed, then the
+    commitment transaction might not confirm until after Alice is able
+    to reclaim her funds from Bob, again creating the opportunity for
+    Bob to end up both paying Mallory and giving a refund to Alice.
+    (This existing issue is what developers were working on fixing when
+    the new issue was discovered.)
+
+    So far we're unaware of any real-world losses due to onchain fee
```

Comment:
At conclusion:
* if node operators feel at risk, they may increase their CLTV_delta (CL: 14, LND: 40, Eclair: 144), it would make attack harder to execute
* but increasing CLTV delta increases offchain cost of liquidity, that's a tradeoff good to underlies

Reply:
A sub-point is that the payer is incentivized to take routes that have a lower total `cltv_delta`.  In C-Lightning we even impose a limit on this total CLTV delta in the built-in `pay` command, so that if the total `cltv_delta` on a route is higher than this limit then we do not even consider that route.  So everyone raising their `cltv_delta`s is likely to also reduce payment reliability overall.